web/themes/new_weather_theme/assets/js/components/combo-box.js

```diff
@@ -119,578 +120,687 @@ const comboTemplate = `
-                    this.cacheLocationGeodata(e.target.dataset.value);
-                });
-                li.addEventListener("mousedown", event => {
-                    // Stop the input from losing focus by
-                    // blocking normal browser behavior here
-                    event.preventDefault();
-                })
-                li.addEventListener("click", this.chooseOption);
-                return li;
-            });
-            // Append to shadow select element
-            this.querySelector('[slot="listbox"]').replaceChildren(...items);
-
-            // If there are results, show the area
-            if(data.suggestions.length){
-                this.showList();
-            } else {
-                this.hideList();
-            }
-            return true;
-        }
-        return false;
+  async getSearchResults(text) {
+    const response = await searchLocation(text);
+    if (response.ok) {
+      const data = await response.json();
+      return data;
+    }
+    return { suggestions: [] };
+  }
+
+  /**
+   * Triggered by input changes.
+   * Will make a request to the ArcGIS endpoint
+   * for search results.
+   * Clears out the current list items and select
+   * options, then creates new versions of each set,
+   * with the correct classes and event handlers
+   * set up.
+   */
+  async updateSearch(text) {
+    const data = await this.getSearchResults(text);
+
+    const makeSectionHeading = (title) => {
+      const heading = document.createElement("li");
+      heading.textContent = title;
+      heading.classList.add("font-mono-sm", "text-base", "padding-1");
+      heading.setAttribute("role", "presentation");
+      return heading;
+    };
+
+    const makeListItem = (suggestion, idx) => {
+      const li = document.createElement("li");
+      li.innerText = suggestion.text;
+      li.setAttribute("role", "option");
+      li.setAttribute("aria-setsize", data.suggestions.length);
+      li.setAttribute("aria-posinset", idx + 1);
+      li.setAttribute("aria-selected", "false");
+      if (suggestion.magicKey) {
+        li.setAttribute("data-value", suggestion.magicKey);
+      } else if (suggestion.url) {
+        li.setAttribute("data-url", suggestion.url);
+      } else {
+        return null;
+      }
+      li.classList.add(...["wx-combo-box__list-option"]);
+      li.id = `${this.id}--item-${idx + 1}`;
+
+      li.addEventListener("focus", (e) => {
+        this.cacheLocationGeodata(e.target.dataset.value);
+      });
+      li.addEventListener("mousedown", (event) => {
+        // Stop the input from losing focus by
+        // blocking normal browser behavior here
+        event.preventDefault();
+      });
+      li.addEventListener("click", this.chooseOption);
+      return li;
+    };
+
+    const saved = this.getSavedResults(text);
+
+    const items = [];
+
+    // If there are saved items, we want to put in a heading list item for them
```

Comment:
This is working well, but wanted to point out that there is a nested list thing happening here where there become two ul lists creating two scrollboxes and I think the outer one might be unnecessary. See the two scrollboxes and the two lists here: 
<img width="507" alt="Screenshot 2024-04-25 at 8 27 28 AM" src="https://github.com/weather-gov/weather.gov/assets/142825699/fdb05073-b3f2-420f-aae2-74669fe3d352">
<img width="369" alt="Screenshot 2024-04-25 at 8 23 17 AM" src="https://github.com/weather-gov/weather.gov/assets/142825699/daeedad8-f21e-433f-99d0-b9f34d69e141">

Reply:
There do need to be two lists for accessibility purposes; that's how we label them separately. But the nested scrollbar is no good. Will look into it!

Reply:
Okay, this was a goof on my part. I gave the container element the same CSS class as the two `ul` elements. The `ul` elements had a max-height and an overflow-y set, but those should have ***only*** been set on the container. I've pushed up what I hope is a fix.